### PR TITLE
updated proposer logic for maxEB

### PR DESCRIPTION
**Pull request by rolfyone**
Added test to show that the consolidation factor affects the proposals in an epoch.

Proposer index 3 went from 0 proposals in epoch 1 up to 4 proposals in epoch 1 after this change (consolidation of 16, so it has 16x32 eth)

## Documentation

- [X] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

**Comment by rolfyone** [comment]
We'll need to update reftests before this will be clean by the looks.

File: ethereum/spec/src/test/java/tech/pegasys/teku/spec/logic/versions/electra/helpers/MiscHelpersElectraTest.java [hunk 1]
```diff
@@ -73,4 +87,45 @@ public void isFormerDepositMechanismDisabled_returnsFalseIfNotDisabled() {
 
     assertThat(miscHelpersElectra.isFormerDepositMechanismDisabled(state)).isFalse();
   }
+
+  @Test
+  void consolidatedValidatorsMoreLikelyToPropose() {
+    final int consolidationAmount = 16;
+    final BeaconState state = randomStateWithConsolidatedValidator(consolidationAmount);
```
Comment (tbenr) [inline ethereum/spec/src/test/java/tech/pegasys/teku/spec/logic/versions/electra/helpers/MiscHelpersElectraTest.java:120]: Don't you mind to have a variable or constant to store the validator index. just to keep those "3"s aligned :)

**Review by tbenr — APPROVED** [review]
LGTM

File: ethereum/spec/src/test/java/tech/pegasys/teku/spec/logic/versions/electra/helpers/MiscHelpersElectraTest.java (hunk 1 above)
Comment (rolfyone) [reply ethereum/spec/src/test/java/tech/pegasys/teku/spec/logic/versions/electra/helpers/MiscHelpersElectraTest.java:120]: sure - thats fine...

**Pull request merged**